rosidl_runtime_py/rosidl_runtime_py/set_message.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2017-2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any
+from typing import Dict
+
+
+class SetFieldError(Exception):
+
+    def __init__(self, field_name, exception):
+        super(SetFieldError, self).__init__()
+        self.field_name = field_name
+        self.exception = exception
+
+
+def set_message_fields(msg: Any, values: Dict[str, str]) -> None:
+    """
+    Set the fields of a ROS message.
+
+    :param msg: The ROS message to populate.
+    :param values: The values to set in the ROS message. The keys of the dictionary represent
+        fields of the message.
+    :raises SetFieldError: If there is an error setting a message field (e.g. the message does
```

Comment:
Instead of always raising a custom exception (which prevents the calling code to distinguish the exceptions type using normal `except X` logic) why not raise specific exceptions for each case. For some even builtin exception types might be appropriate, e.g. `KeyError` when `values` contains a key which is not a field in the message.

Reply:
I've removed the custom exception in 4171c5a, simplifying the implementation.